### PR TITLE
Fix newline addition in create function codeaction and Improve Abstract Code Action Test

### LINKED ISSUE #37343: Create function code action adds a new line unnecessarily to the end of the function
**Description:**
Consider following scenarios in module level and inside function body, both with and without a new line at the end.  We should improve this.

https://user-images.githubusercontent.com/46120162/184291488-bda11270-e14a-4924-b09f-ca84af1f6f10.mov

**Related Issues (optional):**
#33138

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #37343 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java [hunk 1]
```diff
@@ -322,7 +322,9 @@ private Optional<NonTerminalNode> findEnclosingModulePartNode(NonTerminalNode no
     private boolean addNewLineAtEnd(Node enclosingNode) {
         Iterator<Node> iterator = enclosingNode.parent().children().iterator();
         while (iterator.hasNext()) {
-            if (iterator.next().lineRange().startLine().line() == enclosingNode.lineRange().endLine().line() + 1) {
+            Node next = iterator.next();
+            if (next.textRange().length() > 0
+                    && next.lineRange().startLine().line() == enclosingNode.lineRange().endLine().line() + 1) {
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java:327]: Shall we move this method to CodeActionUtil?

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_local_var2.bal [hunk 2]
```diff
@@ -0,0 +1,3 @@
+public function main() {
+    int var1 = createInt();
+}
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_local_var2.bal:3]:
```suggestion
}

```
Shall we add a new line here

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java (hunk 1 above)
Comment (gayalkuruppu) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java:327]: We can write this using a for loop also
```
    private boolean addNewLineAtEnd(Node enclosingNode) {
        for (Node child : enclosingNode.parent().children()) {
            if (child.textRange().length() > 0
                    && child.lineRange().startLine().line() == enclosingNode.lineRange().endLine().line() + 1) {
                return true;
            }
        }
        return false;
    }
    ```

**Review by gayalkuruppu — DISMISSED** [review]
Shall we fix two failing test cases?

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_local_var2.bal (hunk 2 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/test/resources/codeaction/create-function/source/create_function_in_local_var2.bal:3]: This is intentional. This test case was added to verify that the code action does not add a new line when the user have not added a new line

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java:327]: Using an iterator is best performance wise (faster)  than a for-loop. Hence, shall we keep this @gayalkuruppu

**Review by IMS94 — APPROVED** [review]

**Review by mohanvive — APPROVED** [review]

**Pull request merged**